warning: pointer of type ‘void *’ used in arithmetic

Build with `-Wall -Wpedantic`:

```c
In file included from ../src/wl-paste.c:19:
../src/wl-paste.c: In function ‘classify_offer_types’:
../src/types/offer.h:40:49: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
   40 |     mime_type != offer->offered_mime_types.data + \
      |                                                 ^
../src/wl-paste.c:64:5: note: in expansion of macro ‘offer_for_each_mime_type’
   64 |     offer_for_each_mime_type(offer, mime_type) {
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
../src/wl-paste.c: In function ‘selection_callback’:
../src/types/offer.h:40:49: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
   40 |     mime_type != offer->offered_mime_types.data + \
      |                                                 ^
../src/wl-paste.c:191:9: note: in expansion of macro ‘offer_for_each_mime_type’
  191 |         offer_for_each_mime_type(offer, mime_type) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/types/offer.h
+++ b/src/types/offer.h
@@ -37,11 +37,10 @@ struct offer {
 #define offer_for_each_mime_type(offer, mime_type) \
 for ( \
     const char *mime_type = offer->offered_mime_types.data; \
-    mime_type != offer->offered_mime_types.data + \
-                 offer->offered_mime_types.size; \
+    mime_type != (const char *) offer->offered_mime_types.data + \
+                                offer->offered_mime_types.size; \
     mime_type += strlen(mime_type) + 1 \
 )
-
 
 void offer_receive(struct offer *self, const char *mime_type, int fd);
 void offer_destroy(struct offer *self);

